python/ray/tests/runtime_env_validation_unit_test.py

```diff
@@ -0,0 +1,121 @@
+# TODO(hjiang): Move conda related unit test to this file also, after addressing the `
+# yaml` third-party dependency issue.
+
```

Comment:
Let's follow what serve does here (`python/ray/serve/tests/unit`) and create `python/ray/tests/unit`. Also checkout out `python/ray/serve/tests/unit/BUILD`. The rule is everything under `unit` should have no `import ray`.

Reply:
If we take a look at the so-called unit test `python/ray/serve/tests/unit`, it's still importing `ray`, so still integration test sadly :(

Reply:
But yeah, I could make another folder for unit test only, SGTM :)